astropy/utils/misc.py

```diff
@@ -159,22 +163,28 @@ class NumpyRNGContext:
 
     """
 
-    def __init__(self, seed):
+    def __init__(self, seed: int) -> None:
```

Comment:
`numpy.random.seed` can accept more than an int. 
https://numpy.org/doc/stable/reference/random/generated/numpy.random.seed.html isn't very specific but at least None also works.

Reply:
`NumpyRNGContext` is not obligated to accept everything `numpy.random.seed()` can accept. Furthermore, `numpy.random.seed()` is legacy API anyways (#11687), so I don't see much value in improving `NumpyRNGContext` even if someone does come complaining that it is too restrictive.